### PR TITLE
Frontend search

### PR DESCRIPTION
**Pull request by Qmpzlawasd**
Added search page

### CODE REVIEW

File: frontend/src/pages/search.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    title: "Saving gold for my rretirem",
+    author: "Me",
+    rating: 4.25,
+    coverImageUrl: "https://edit.org/images/cat/book-covers-big-2019101610.jpg",
+  },
+];
+
+export default function SearchPage() {
+  let MIN_BOOK_YEAR = 1500;
+  let MAX_BOOK_YEAR = 2023;
```
Comment (tezkerek) [inline frontend/src/pages/search.tsx:89]: `const`, move out of component.

File: frontend/src/pages/search.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  const [maxYear, setMaxYear] = useState<number>(2023);
+  const [minYear, setMinYear] = useState<number>(1984);
+  const [query, setQuery] = useState<string>("");
+  const submitToBE = () => {
+    console.log(query, maxYear, minYear, selectedPublishers);
+  };
+
+  return (
+    <>
+      <Head>
```
Comment (tezkerek) [inline frontend/src/pages/search.tsx:100]: Missing `<title></title>`

File: frontend/src/pages/search.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            <input
+              className={styles.inputField}
+              type="text"
+              value={query}
+              onChange={(e) => {
+                setQuery(e.target.value);
+              }}
+            />
+            <button className={styles.submitButton} type="submit">
+              Meow!
```
Comment (tezkerek) [inline frontend/src/pages/search.tsx:118]: `s/Meow!/Search`

File: frontend/src/pages/search.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+            min={MIN_BOOK_YEAR}
+            css={sliderColor}
+            value={maxYear}
+            aria-label="Default"
+            valueLabelDisplay="auto"
+            onChange={(event: Event, newValue: number | number[]) => {
+              newValue = newValue as number;
+              if (newValue < minYear) {
+                setMaxYear(minYear);
+              } else setMaxYear(newValue);
```
Comment (tezkerek) [inline frontend/src/pages/search.tsx:143]: Shorter: `setMaxYear(Math.max(newValue as number, minYear))`

File: frontend/src/pages/search.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            max={MAX_BOOK_YEAR}
+            min={MIN_BOOK_YEAR}
+            value={minYear}
+            aria-label="Big"
+            valueLabelDisplay="auto"
+            onChange={(event: Event, newValue: number | number[]) => {
+              newValue = newValue as number;
+              if (newValue > maxYear) {
+                setMinYear(maxYear);
+              } else setMinYear(newValue);
```
Comment (tezkerek) [inline frontend/src/pages/search.tsx:159]: Same as above

File: frontend/src/pages/search.tsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+          <p className={styles.filterTitle}>Choose Publisher</p>
+          <PublisherFilter
+            publishers={mockPublishers}
+            selectedPublishers={selectedPublishers}
+            onChange={(publishers: Array<Publisher>) =>
+              setSelectedPublishers(publishers)
+            }
+          />
+          <p className={styles.filterTitle}>Choose Max Year</p>
+          <Slider
```
Comment (tezkerek) [inline frontend/src/pages/search.tsx:132]: I would inverse the two sliders: min should be above max.

File: frontend/src/styles/Search.module.scss [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    display: flex;
+    align-items: center;
+    grid-area: search;
+    padding-left: 3%;
+    .inputField {
+        outline: none;
+        border-radius: 0%;
+        font-size: 1.1em;
+        width: 54vw;
+        padding:4px;
```
Comment (tezkerek) [inline frontend/src/styles/Search.module.scss:25]: Run the formatter in this file.

File: frontend/src/components/filters.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      );
+    }
+    onChange(selectedPublishers);
+  };
+
+  return (
+    <>
+      {publishers.map((publisher, i) => (
+        <div key={i}>
+          <input css={checkboxColor} type="checkbox" onChange={handleBox} value={publisher} />
```
Comment (tezkerek) [inline frontend/src/components/filters.tsx:33]: Checked state should be set based on whether the publisher is selected or not.

File: frontend/src/components/filters.tsx
Comment (tezkerek) [inline frontend/src/components/filters.tsx:1]: Rename file to `PublisherFilter.tsx`

File: frontend/src/components/filters.tsx [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+          <label css={labelCheckbox}>{publisher}</label>
+        </div>
+      ))}
+    </>
+  );
+}
+const checkboxColor = css`
+    accent-color: var(--color-accent);
+`
+const labelCheckbox= css`
```
Comment (tezkerek) [inline frontend/src/components/filters.tsx:43]: Forgot to run formatter.

**Review by tezkerek — CHANGES_REQUESTED** [review]

File: frontend/src/components/filters.tsx (hunk 8 above)
Comment (Qmpzlawasd) [reply frontend/src/components/filters.tsx:33]: > Checked state should be set based on whether the publisher is selected or not.

Could you elaborate?

Comment (tezkerek) [reply frontend/src/components/filters.tsx:33]: Currently the selectedPublishers list and the checkboxes aren't synchronized. They only coincide when the initial selectedPublishers list is empty. What if we started with `selectedPublishers = ["Alcatel"]`? All checkboxes would still be empty.

Each checkbox input should have `defaultChecked={is this publisher selected?}`.

Comment (tezkerek) [reply frontend/src/components/filters.tsx:33]: I think `checked={is publisher selected?}` would be better, actually.

**Review by tezkerek — CHANGES_REQUESTED** [review]

**Review by tezkerek — APPROVED** [review]

**Pull request merged**